include/xcshell/error_handling.h

```diff
@@ -0,0 +1,18 @@
+#ifndef INCLUDE_XCSHELL_ERROR_HANDLING_H_
+#define INCLUDE_XCSHELL_ERROR_HANDLING_H_
+
+#include <iostream>
+
+enum ErrorType { FATAL_ERROR = 1, NORMAL_ERROR = 2 };
```

Comment:
Enum class, I mean, not the old style enum. You can check out this link for more information:
https://www.geeksforgeeks.org/enum-classes-in-c-and-their-advantage-over-enum-datatype/

Reply:
Also, you can put this enum class inside class ErrorHandling. For example:


```c++
class ErrorHandling {
public:
  enum class Error Type {...}
  
  static void PrintSystemError(std::ostream& os_err);
   static int ErrorDispatchHandler(int return_value, ErrorType error_type);
}
```